lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx

```diff
@@ -5,160 +5,489 @@
-      }
-  );
-
-  it('sets the NxTransferListHalf footerContent to "x items Added" where x is the number of addedItems by default',
-      function() {
-        const noItems: DataItem[] = [],
-            oneItem = [{ id: 'a', displayName: 'a' }],
-            twoItems = [{ id: 'a', displayName: 'a' }, { id: 'b', displayName: 'b' }],
-            noItemComponent = getShallow({ addedItems: noItems }),
-            oneItemComponent = getShallow({ addedItems: oneItem }),
-            twoItemComponent = getShallow({ addedItems: twoItems });
-
-        expect(noItemComponent.find(NxTransferListHalf)).toHaveProp('footerContent', '0 items Added');
-        expect(oneItemComponent.find(NxTransferListHalf)).toHaveProp('footerContent', '1 item Added');
-        expect(twoItemComponent.find(NxTransferListHalf)).toHaveProp('footerContent', '2 items Added');
-      }
-  );
-
-  it('calls onRemove without moved items when the NxTransferListHalf onMoveAll is fired', function() {
-    const onRemove = jest.fn(),
-        component = getShallow({
-          onRemove,
-          addedItems: [{ id: 'c', displayName: 'a' }, { id: 'B', displayName: 'b' }, { id: 'C', displayName: 'c' }]
+        it('renders matches as type="button" and an accessible name', function() {
+          const matches = viewWithMatches().getAllByRole('menuitem');
+
+          expect(matches[0]).toHaveAttribute('type', 'button');
+          expect(matches[0]).toHaveAccessibleName('Item 1');
+
+          expect(matches[1]).toHaveAttribute('type', 'button');
+          expect(matches[1]).toHaveAccessibleName('Item 2');
         });
 
-    expect(onRemove).not.toHaveBeenCalled();
+        it('calls onSearchMatchSelect when user selects a match', async function() {
+          const user = userEvent.setup(),
+              onSearchMatchSelect = jest.fn().mockImplementation((_, evt) => { evt.persist(); }),
+              view = viewWithMatches({ onSearchMatchSelect }),
+              firstMatch = view.getAllByRole('menuitem')[0];
+
+          expect(onSearchMatchSelect).not.toHaveBeenCalled();
+
+          await user.click(firstMatch);
+
+          expect(onSearchMatchSelect)
+              .toHaveBeenCalledWith({ id: '1', displayName: 'Item 1'}, expect.objectContaining({ target: firstMatch }));
+        });
+      });
+
+      describe('when there are no matches', function() {
+        const viewNoMatches = (props?: Partial<Props>) => quickRender({ searchText: 'i', ...props });
+
+        it('renders a dropdown with role=alert when in empty, or loading states', function() {
+          expect(viewNoMatches().getByRole('alert')).toBeInTheDocument();
+          expect(viewNoMatches({ loading: true }).getByRole('alert')).toBeInTheDocument();
+        });
+
+        it('renders a dropdown with no role when in an error state', function() {
+          const dropdown = renderEl({ loadError: 'oops' })!.querySelector('.nx-search-dropdown__menu--error');
```

Comment:
Maybe we can test the non-existence of the role instead?

Reply:
> Maybe we can test the non-existence of the role instead?

I had left this as a query selector since I figured it was best to test that the dropdown doesn't have either the menu or alert role. But if it's in an alert state, the dropdown contains an alert, so `getByRole('alert')` would return true. However, I'm happy to change it as well and just assert that role=menu isn't present!

Reply:
> > Maybe we can test the non-existence of the role instead?
> 
> I had left this as a query selector since I figured it was best to test that the dropdown doesn't have either the menu or alert role. But if it's in an alert state, the dropdown contains an alert, so `getByRole('alert')` would return true. However, I'm happy to change it as well and just assert that role=menu isn't present!

Yeah I think it's better to change it so that we don't query internal class details. We'd also be consistent with other tests where we're not as concerned with implementation details.

Reply:
> > > Maybe we can test the non-existence of the role instead?
> > 
> > 
> > I had left this as a query selector since I figured it was best to test that the dropdown doesn't have either the menu or alert role. But if it's in an alert state, the dropdown contains an alert, so `getByRole('alert')` would return true. However, I'm happy to change it as well and just assert that role=menu isn't present!
> 
> Yeah I think it's better to change it so that we don't query internal class details. We'd also be consistent with other tests where we're not as concerned with implementation details.

Makes sense, added here: [384b76e](https://github.com/sonatype/sonatype-react-shared-components/pull/945/commits/384b76eb3390b70799304e36d7a6935fb679b883)